controlplane/eks/api/v1alpha4/awsmanagedcontrolplane_webhook.go

```diff
@@ -131,6 +131,13 @@ func (r *AWSManagedControlPlane) ValidateUpdate(old runtime.Object) error {
 		)
 	}
 
+	// If encryptionConfig is already set, do not allow removal of it.
+	if oldAWSManagedControlplane.Spec.EncryptionConfig != nil && r.Spec.EncryptionConfig == nil {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "encryptionConfig"), r.Spec.EncryptionConfig, "field cannot be set to nil"),
```

Comment:
It may be good to give a more user friendly message. Something like "disabling EKS encryption is not allowed after it has been enabled".